Code Broken : HerokuDatabaseConfiguration

Using jhipster 2.27.0 apoint to master, the code generated for 'yo jhipster:heroku' create the class HerokuDatabaseConfiguration using jHipsterProperties.getDatasource() in the code, but this method was removed.

``` JAVA
Error: Command failed: /bin/sh -c ./gradlew -Pprod bootRepackage -x test && heroku deploy:jar --jar build/libs/*.war
npm WARN optional Skipping failed optional dependency /chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.0.7
/mnt/dados/projetos/comment/codigo/api/src/main/java/me/comment/api/config/HerokuDatabaseConfiguration.java:30: error: cannot find symbol
                config.addDataSourceProperty("cachePrepStmts", jHipsterProperties.getDatasource().isCachePrepStmts());
                                                                                 ^
  symbol:   method getDatasource()
  location: variable jHipsterProperties of type JHipsterProperties
/mnt/dados/projetos/comment/codigo/api/src/main/java/me/comment/api/config/HerokuDatabaseConfiguration.java:31: error: cannot find symbol
                config.addDataSourceProperty("prepStmtCacheSize", jHipsterProperties.getDatasource().getPrepStmtCacheSize());
                                                                                    ^
  symbol:   method getDatasource()
  location: variable jHipsterProperties of type JHipsterProperties
/mnt/dados/projetos/comment/codigo/api/src/main/java/me/comment/api/config/HerokuDatabaseConfiguration.java:32: error: cannot find symbol
                config.addDataSourceProperty("prepStmtCacheSqlLimit", jHipsterProperties.getDatasource().getPrepStmtCacheSqlLimit());
                                                                                        ^
  symbol:   method getDatasource()
  location: variable jHipsterProperties of type JHipsterProperties
3 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

```

--- a/generators/heroku/templates/src/main/java/package/config/_HerokuDatabaseConfiguration.java
+++ b/generators/heroku/templates/src/main/java/package/config/_HerokuDatabaseConfiguration.java
@@ -1,10 +1,11 @@
 package <%=packageName%>.config;
 
-import com.zaxxer.hikari.HikariConfig;
 import com.zaxxer.hikari.HikariDataSource;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties;<% if (hibernateCache == 'hazelcast') { %>
+import org.springframework.boot.autoconfigure.jdbc.DataSourceBuilder;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;<% if (hibernateCache == 'hazelcast') { %>
 import org.springframework.cache.CacheManager;<% } %>
 import org.springframework.context.ApplicationContextException;
 import org.springframework.context.annotation.*;
@@ -18,23 +19,17 @@ public class HerokuDatabaseConfiguration {
     private final Logger log = LoggerFactory.getLogger(HerokuDatabaseConfiguration.class);
 
     @Bean
-    public DataSource dataSource(DataSourceProperties dataSourceProperties, JHipsterProperties jHipsterProperties<% if (hibernateCache == 'hazelcast') { %>, CacheManager cacheManager<% } %>) {
+    @ConfigurationProperties(prefix = "spring.datasource.hikari")
+    public DataSource dataSource(DataSourceProperties dataSourceProperties<% if (hibernateCache == 'hazelcast') { %>, CacheManager cacheManager<% } %>) {
         log.debug("Configuring Heroku Datasource");
 
         String herokuUrl = System.getenv("JDBC_DATABASE_URL");
         if (herokuUrl != null) {
-	    HikariConfig config = new HikariConfig();
-
-	    //MySQL optimizations, see https://github.com/brettwooldridge/HikariCP/wiki/MySQL-Configuration
-	    if ("com.mysql.jdbc.jdbc2.optional.MysqlDataSource".equals(dataSourceProperties.getDriverClassName())) {
-                config.addDataSourceProperty("cachePrepStmts", jHipsterProperties.getDatasource().isCachePrepStmts());
-                config.addDataSourceProperty("prepStmtCacheSize", jHipsterProperties.getDatasource().getPrepStmtCacheSize());
-                config.addDataSourceProperty("prepStmtCacheSqlLimit", jHipsterProperties.getDatasource().getPrepStmtCacheSqlLimit());
-            }
-
-            config.setDataSourceClassName(dataSourceProperties.getDriverClassName());
-            config.addDataSourceProperty("url", herokuUrl);
-            return new HikariDataSource(config);
+            return DataSourceBuilder
+                .create(dataSourceProperties.getClassLoader())
+                .type(HikariDataSource.class)
+                .url(herokuUrl)
+                .build();
         } else {
             throw new ApplicationContextException("Heroku database URL is not configured, you must set $JDBC_DATABASE_URL");
         }

--- a/generators/server/templates/src/main/java/package/config/_DatabaseConfiguration.java
+++ b/generators/server/templates/src/main/java/package/config/_DatabaseConfiguration.java
@@ -4,7 +4,6 @@ import <%=packageName%>.config.liquibase.AsyncSpringLiquibase;
 
 import com.codahale.metrics.MetricRegistry;
 import com.fasterxml.jackson.datatype.hibernate4.Hibernate4Module;
-import com.zaxxer.hikari.HikariConfig;
 import com.zaxxer.hikari.HikariDataSource;
 import liquibase.integration.spring.SpringLiquibase;<% } %><% if (databaseType == 'mongodb' && authenticationType == 'oauth2') { %>
 import <%=packageName%>.config.oauth2.OAuth2AuthenticationReadConverter;<% } %><% if (databaseType == 'mongodb') { %>

